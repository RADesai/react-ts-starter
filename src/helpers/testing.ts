import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

export const matchSnapshot = (component: React.ReactElement<any, string | React.JSXElementConstructor<any>>) => {
    const tree = renderer
        .create(component)
        .toJSON();
    expect(tree).toMatchSnapshot();
}

export const hasText = (text: string) => {
    const linkElement = screen.getByText(text);
    expect(linkElement).toBeInTheDocument();
}
