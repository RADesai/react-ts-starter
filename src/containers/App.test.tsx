import { render } from '@testing-library/react';

import App from './App';
import { hasText, matchSnapshot } from '../helpers/testing'

test('renders app template', () => {
  render(<App />);

  hasText('React - TypeScript - Tailwind Starter Kit')
  matchSnapshot(<App />)
});
