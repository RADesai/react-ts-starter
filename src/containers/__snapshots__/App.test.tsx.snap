// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app template 1`] = `
<div
  className="App"
>
  <div
    className="text-3xl font-bold underline"
  >
    React - TypeScript - Tailwind Starter Kit
  </div>
</div>
`;
